src/common/providers/https.ts

```diff
@@ -722,6 +744,36 @@ function wrapOnCallHandler<Req = any, Res = any>(
   version: "gcfv1" | "gcfv2"
 ): (req: Request, res: express.Response) => Promise<void> {
   return async (req: Request, res: express.Response): Promise<void> => {
+    const abortController = new AbortController();
+    let heartbeatInterval: NodeJS.Timeout | null = null;
+
+    const heartbeatSeconds =
+      options.heartbeatSeconds === undefined ? DEFAULT_HEARTBEAT_SECONDS : options.heartbeatSeconds;
+
+    const clearScheduledHeartbeat = () => {
+      if (heartbeatInterval) {
+        clearTimeout(heartbeatInterval);
+        heartbeatInterval = null;
+      }
+    };
+
+    const scheduleHeartbeat = () => {
+      clearScheduledHeartbeat();
+      if (!abortController.signal.aborted) {
+        heartbeatInterval = setTimeout(() => {
```

Comment:
Why not use setInterval()?

Reply:
Using `setTimeout` instead since I'm trying ton control more precisely whether next "tick" should happen or not. That's why this call is recursive.